src/modules/Dropdown/Dropdown.js

```diff
@@ -481,7 +490,7 @@ export default class Dropdown extends Component {
     debug('handleBlur()')
     const { onBlur } = this.props
     if (onBlur) onBlur(e)
-    this.setState({ focus: false })
+    this.setState({ focus: false, blurEvent: e })
```

Comment:
Because of [React's event pooling](https://facebook.github.io/react/docs/events.html#event-pooling) there's a chance this event will be nullified by the time it is passed to `selectHighlightedItem`.  Instead of putting it on state, and passing it later, let's just pass it right here like `selectOnEnter` does:

```js
handleBlur = (e) => {
  debug('handleBlur()')
  const { multiple, onBlur, selectOnBlur } = this.props
  if (onBlur) onBlur(e)
  if (selectOnBlur && !multiple) this.selectHighlightedItem(e)
  this.setState({ focus: false })
}
```

Then we can remove this from `componentDidUpdate`:

```diff
-if (this.props.selectOnBlur && !this.props.multiple) this.selectHighlightedItem(this.state.blurEvent)
```